app/controllers/application_controller.rb

```diff
@@ -784,9 +790,12 @@ def build_not_found_page(status = 404, layout = false)
     end
 
     @container_class = "wrap not-found-container"
-    @slug = (params[:slug].presence || params[:id].presence || "").tr('-', '')
+    @slug = (params[:slug].presence || params[:id].presence || "").tr('-', ' ')
```

Comment:
Why replace the dash (`-`) by space (` `) in the slug?

Reply:
Let's take [an example](https://meta.discourse.org/t/discourse-2-4-0-beta5-release-notes/12998810).

The search input field has all the words in the slug concatenated.